basics/Main.java

```diff
@@ -17,4 +22,21 @@ public static String pluralize(String animal, int count) {
     }
     return plural;
   }
+
+  public static void flipNHeads(int num){
+    int flips = 0;
+    int numOfHeads = 0;
+    while(numOfHeads < num){
+      double randomNum = Math.random();
```

Comment:
When you can, declare variable outside of loops. That way, instead of having to recreate a variable each time it loops you just reassign it each time it loops saving. Having said all of that, the increase in efficiency by doing this probably isn't much...